packages/p2p/src/stores/sendbird-store.ts

```diff
@@ -295,44 +295,6 @@ export default class SendbirdStore extends BaseStore {
         }
     }
 
-    onMessagesScroll() {
-        if (this.scroll_debounce) {
-            clearInterval(this.scroll_debounce);
-        }
-
-        this.scroll_debounce = setTimeout(() => {
-            if (!this.messages_ref?.current) return;
-
-            if (this.messages_ref.current.scrollTop === 0) {
-                this.setIsChatLoading(true);
-                const oldest_message_timestamp = this.chat_messages.reduce(
-                    (prev_created_at, chat_message) =>
-                        chat_message.created_at < prev_created_at ? chat_message.created_at : prev_created_at,
-                    Infinity
-                );
-                this.getPreviousMessages(oldest_message_timestamp)
-                    .then(chat_messages => {
-                        if (chat_messages && chat_messages.length > 0) {
-                            const previous_messages = chat_messages.map(chat_message =>
-                                convertFromChannelMessage(chat_message)
-                            );
-
-                            this.replaceChannelMessage(0, 0, previous_messages[0]);
-                        }
-                        this.setIsChatLoading(false);
-                    })
-                    .catch(error => {
-                        // eslint-disable-next-line no-console
-                        console.warn(error);
-                    });
-            } else {
-                (async () => {
-                    await this.markMessagesAsRead(true);
-                })();
-            }
-        }, 1000);
-    }
-
```

Comment:
The reason why I removed this function is because there are already other functions that help to either `mark the messages as read` or `getting the messages from sendbird`. The reason why we got the bug where it would reload the chat and scroll to the botton is because of the if check where it checks if scrollTop is 0. And we would then get all the previous messages from there. 

But we shouldn't do that as inside `registerMobxReactions` we already call `initialiseOrderMessages` where it gets all the previous messages already and populates the chat screen with the messages.

And for marking the messages as read, we already have another function that does this which is inside `registerEventListeneres` where it checkd for any focus event onto the chat screen and calles `markMessagesAsRead`.

Reply:
so will this work if there are more than 50 messages? is it working fine without this function?

Reply:
ah i just checked again, its not getting the previous message, i think ill change way we do the message on scroll instead

Reply:
Did this fixed?